contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {ILens} from "./ILens.sol";
+import {IMorpho} from "./IMorpho.sol";
+import {CoreRef} from "../../refs/CoreRef.sol";
+import {PCVDeposit} from "../PCVDeposit.sol";
+import {ICompoundOracle, ICToken} from "./ICompound.sol";
+
+/// @notice PCV Deposit for Morpho-Compound V2.
+/// Implements the PCV Deposit interface to deposit and withdraw funds in Morpho
+/// Liquidity profile of Morpho for this deposit is fully liquid for USDC and DAI
+/// because the incentivized rates are higher than the P2P rate.
+/// Only for depositing USDC and DAI. USDT is not in scope.
+/// @dev approves the Morpho Deposit to spend this PCV deposit's token,
+/// and then calls supply on Morpho, which pulls the underlying token to Morpho,
+/// drawing down on the approved amount to be spent,
+/// and then giving this PCV Deposit mTokens in exchange for the underlying
+contract MorphoCompoundPCVDeposit is PCVDeposit {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the lens contract for morpho-compound v2
+    address public constant LENS = 0x930f1b46e1D081Ec1524efD95752bE3eCe51EF67;
+
+    /// @notice reference to the morpho-compound v2 market
+    IMorpho public constant MORPHO =
+        IMorpho(0x8888882f8f843896699869179fB6E4f7e3B58888);
+
+    /// @notice reference to underlying token
+    address public immutable token;
+
+    /// @notice cToken in compound this deposit tracks
+    /// used to inform morpho about the desired market to supply liquidity
+    address public immutable cToken;
+
+    constructor(address _core, address _cToken) CoreRef(_core) {
+        cToken = _cToken;
+        token = ICToken(_cToken).underlying();
```

Comment:
beware this does not work with the cETH token, so if you want to handle ETH I suggest you to change this part of the code.